rstar/src/algorithm/nearest_neighbor.rs

```diff
@@ -272,29 +272,26 @@ where
             }
             RTreeNodeDistanceWrapper {
                 node: RTreeNode::Leaf(ref t),
-                ..
+                distance,
             } => {
-                return Some(t);
+                return Some((t, distance));
             }
         }
     }
     None
 }
 
-pub fn nearest_neighbors<T>(
+#[allow(clippy::type_complexity)]
+pub fn nearest_neighbors_with_distance_2<T>(
     node: &ParentNode<T>,
     query_point: <T::Envelope as Envelope>::Point,
-) -> Vec<&T>
+) -> Option<(Vec<&T>, <<T::Envelope as Envelope>::Point as Point>::Scalar)>
```

Comment:
I think Clippy is on the money here. Maybe you could a local generic type alias like

```rust
type Distance<T> = <<T::Envelope as Envelope>::Point as Point>::Scalar where T: PointDistance;
```

and use it throughout this module?